src/parser/EBNF.md

```diff
@@ -0,0 +1,23 @@
+```ebnf
+machine_definition = "machine", "(", params_list ,")", [ "(", params_list ,")" ], machine_block;
```

Comment:
> Also, it seems `( param_list )` appears twice, it should only appear once right?

The second `( param_list )` is the return type.  There's no `-> ( param_list )` because `->` is already used for state transition.